Support a comma separated list of job name filters.

harshal opened the pull request:
@jainpawan  @pk026  This is totally *completely* untested. Please check if something like this will work for us. 

The idea is pretty simple :  Consider all those jobs where you get only a small number each day (say < 100), but you cannot tolerate latency i.e. you have to run them every few seconds. For these it will be much more efficient to just run them together. You will not have to pay the cost of django setup/teardown for each job. The number of processes running together and the memory used as a result should also go down drastically. 

I see a couple of downsides:
- The meaning of the -j param changes a little bit. Now the limit applies to each individual filter in the list
- Potentially, you are somewhat de-prioritizing the jobs at the end of the list v/s those in the front. We can probably live with that for infrequent jobs.

harshal committed:
Support a comma separated list of job name filters.

harshal commented:
@jainpawan : The reason I did not bother pushing things down into sql is that I don't know how to do something like 
`WHERE NAME starts with [...list of names...] `
using django ORM. Basically how to combine LIKE name% and IN. If we solve that, then that would be better of course.

jainpawan commented:
we can do something like..

from django.db.models import Q
filter = Q(True)
for name in names:
    filter |= Q(name__startswith=name)

harshal committed:
Refactor the code so that the base queryset that filters by job names
and other basic criteria is in a separate variable. This makes
the next patch easier

harshal commented:
> we can do something like..
> 
> from django.db.models import Q
> filter = Q(True)
> for name in names:
> filter |= Q(name__startswith=name)

Thanks for the hint! I have updated the PR.

jainpawan commented:
@pk026 Lets discuss how do we roll this out quickly..
